api/python/provisioner/commands/create_user.py

```diff
@@ -0,0 +1,365 @@
+
+        logger.info(f"Creating new user: {uname}")
+        home_dir = SEAGATE_USER_HOME_DIR / uname
+        ssh_dir = home_dir / '.ssh'
+
+        user_fileroots_dir = Path(
+            PRVSNR_FILEROOT_DIR /
+            SEAGATE_USER_FILEROOT_DIR_TMPL.format(uname=uname)
+        )
+
+        keyfile = user_fileroots_dir / f'id_rsa_{uname}'
+        keyfile_pub = keyfile.with_name(f'{keyfile.name}.pub')
+
+        # nodes = PillarKey('cluster/node_list')
+
+        # nodelist_pillar = PillarResolver(LOCAL_MINION).get([nodes])
+        # nodelist_pillar = next(iter(nodelist_pillar.values()))
+
+        local_minion = local_minion_id()
+        cluster_path = PillarKey('cluster')
+        get_result = PillarResolver(local_minion).get([cluster_path])
+        cluster_pillar = get_result[local_minion][cluster_path]
+        nodes_list = [
+            node for node in cluster_pillar.keys()
+            if 'srvnode-' in node
+        ]
+
+        if (not cluster_pillar or
+                cluster_pillar is values.MISSED):
+            raise BadPillarDataError(
+                'value for {} is not specified'.format(cluster_path.pi_key)
+            )
+
+        def _prepare_user_fileroots_dir():
+            StateFunExecuter.execute(
+                'file.directory',
+                fun_kwargs=dict(
+                    name=str(user_fileroots_dir),
+                    makedirs=True
+                )
+            )
+
+        def _generate_ssh_keys():
+            StateFunExecuter.execute(
+                'cmd.run',
+                fun_kwargs=dict(
+                    name=(
+                        f"ssh-keygen -f {keyfile} "
+                        "-q -C '' -N '' "
+                        "-t rsa -b 4096 <<< y"
+                    )
+                )
+            )
+            StateFunExecuter.execute(
+                'ssh_auth.present',
+                fun_kwargs=dict(
+                    # name param is mandetory and expects ssh key
+                    # but ssh key is passed as source file hence name=None
+                    name=None,
+                    user=uname,
+                    source=str(keyfile_pub),
+                    config=str(user_fileroots_dir / 'authorized_keys')
+                )
+            )
+
+        def _generate_ssh_config():
+            for node in nodes_list:
+                hostname = PillarKey(
+                    'cluster/' + node + '/hostname'
+                )
+
+                hostname_pillar = PillarResolver(LOCAL_MINION).get([hostname])
+                hostname_pillar = next(iter(hostname_pillar.values()))
+
+                if (not hostname_pillar[hostname] or
+                        hostname_pillar[hostname] is values.MISSED):
+                    raise BadPillarDataError(
+                        'value for {} is not specified'.format(hostname.pi_key)
+                    )
+
+                ssh_config = f'''Host {node} {hostname_pillar[hostname]}
```

Comment:
We might want to make this optional.
Add a TODO here

Reply:
Addressed